lib/net/ldap/connection.rb

```diff
@@ -8,56 +8,49 @@ class Net::LDAP::Connection #:nodoc:
 
   def initialize(server)
     @instrumentation_service = server[:instrumentation_service]
-    server[:hosts] = [[server[:host], server[:port]]] if server[:hosts].nil?
 
     if server[:socket]
-      prepare_socket(server)
+      prepare_socket(server[:socket], server[:encryption])
     else
-      open_connection(server)
+      hosts = server[:hosts]
+      hosts = [[server[:host], server[:port]]] if hosts.nil?
+      open_connection(hosts, server[:encryption])
```

Comment:
After looking into this, I think it won't work out.  Most of what `open_connection` does in this patch is manage exception dispatching, and it needs to do this because in the multi-host case it must suppress any exceptions raised if a connection is successfully opened.  Opening the connection is actually handled by `prepare_socket` and `TCPSocket.new`, one line in the `open_connection` method.  The rest of the method is the looping over hosts to try, gathering of exceptions, and finally raising an exception based on the captured exceptions if no connection was opened.

In other words, to do what you suggest moves about 19 lines from `open_connection` into `initialize`, leaving `open_connection` as a one-liner and `initialize` rather more bloated.  I don't really see how this would address the exception handling code smell either since we're just moving logic around rather than eliminating or refactoring it significantly.

Have you looked at the result of the patch rather than the diffs yet?  I think you'll see that the patch is noisy only because it's simplifying the code.